packages/kit/src/core/prerender/prerender.js

```diff
@@ -381,6 +407,17 @@ export async function prerender() {
 
 	await q.done();
 
+	// handle invalid fragment links
+	for (const [key, referrers] of expected_deeplinks) {
+		const [path, id] = key.split('#');
+
+		const actual_ids = actual_deeplinks.get(path);
+
+		if (actual_ids?.includes(id)) {
+			handle_missing_id({ id, path, referrers: Array.from(referrers) });
```

Comment:
printing those missing ids in a loop like this means that it throws after the first error by default, and you don't see the other wrong links. I think it would be better to print all out in one go.
Aside: It would be good to also add something like "you can adjust the handling of missing ids through -config option-" so people have a clearer way out in case they don't want to change the hash.

Reply:
Had an internal debate with myself on this — initially I could have gone either way, but I think I lean towards keeping it as-is:

* we fail eagerly for HTTP errors so it might be odd not to do the same thing here
* the custom function signature gets a bit trickier, and we'd need to call it `handleMissingIds`. it's a bit unusual for an error handler to handle multiple errors in one go
* given that the default behaviour is to fail, it would be unlikely for multiple missing IDs to be introduced in one go (the exception being the day this feature ships and previously hidden bad links surface en masse)
* if someone does need to see all the missing IDs together, they could do `handleMissingId: warn` and run a build